crates/proof-of-sql/src/sql/proof/proof_plan.rs

```diff
@@ -46,6 +46,21 @@ pub trait ProofPlan: Debug + Send + Sync + ProverEvaluate {
 
     /// Return all the tables referenced in the Query
     fn get_table_references(&self) -> IndexSet<TableRef>;
+
+    /// Return the row number range of tables referenced in the Query
+    ///
+    /// Basically we are looking for the smallest offset and the largest offset + length
+    /// so that we have an index range of the table rows that the query is referencing.
+    fn get_index_range(&self, accessor: &dyn MetadataAccessor) -> (usize, usize) {
+        self.get_table_references().iter().fold(
+            (usize::MAX, 0),
```

Comment:
* What happens if there are no tables referenced? I would make the output optional rather than have an incorrect default.
* I would prefer to see this take `table_references` as an argument rather than `self`. That way, `table_references` can be used outside this method multiple times if needed without needing to call `get_table_references` twice.

Reply:
@JayWhite2357 If there are no tables referenced the amount of rows here is 1 since the only possible situation is `EmptyExec`. Max of intermediate `ProofPlan`s can be higher with unions but that requires more refactoring (mostly replacement of accessors with tables in `result_evaluate` and `final_round_evaluate`.